Source/WebCore/style/StyleInterpolation.cpp

```diff
@@ -0,0 +1,386 @@
+{
+    Context context { customProperty, progress, false, compositeOperation, iterationCompositeOperation, currentIteration, client };
+
+    auto [fromValue, toValue] = customPropertyValuesForInterpolation(customProperty, from, to);
+    if (!fromValue || !toValue)
+        return;
+
+    bool isInherited = client.document()->customPropertyRegistry().isInherited(customProperty);
+    destination.setCustomPropertyValue(interpolatedCSSCustomPropertyValue(from, to, *fromValue, *toValue, context), isInherited);
+}
+
+static bool syntaxValuesRequireInterpolationForAccumulativeIteration(const CSSCustomPropertyValue::SyntaxValue& a, const CSSCustomPropertyValue::SyntaxValue& b, bool isList)
+{
+    return WTF::switchOn(a,
+        [b, isList](const WebCore::Length& aLength) {
+            ASSERT(std::holds_alternative<WebCore::Length>(b));
+            return !isList && lengthsRequireInterpolationForAccumulativeIteration(aLength, std::get<WebCore::Length>(b));
+        },
+        [](const RefPtr<TransformOperation>&) {
+            return true;
+        },
+        [](const Style::Color&) {
+            return true;
+        },
+        [](auto&) {
+            return false;
+        }
+    );
+}
+
+static bool typeOfSyntaxValueCanBeInterpolated(const CSSCustomPropertyValue::SyntaxValue& syntaxValue)
+{
+    return WTF::switchOn(syntaxValue,
+        [](const WebCore::Length&) {
+            return true;
+        },
+        [](const Color&) {
+            return true;
+        },
+        [](CSSCustomPropertyValue::NumericSyntaxValue) {
+            return true;
+        },
+        [](const CSSCustomPropertyValue::TransformSyntaxValue&) {
+            return true;
+        },
+        [](RefPtr<StyleImage>) {
+            return false;
+        },
+        [](auto&) {
+            return false;
+        }
+    );
+}
+
+// MARK: - Exposed functions
+
+bool isAdditiveOrCumulative(const AnimatableCSSProperty& property)
+{
+    return WTF::switchOn(property,
+        [](CSSPropertyID propertyId) {
+            return !CSSProperty::animationUsesNonAdditiveOrCumulativeInterpolation(propertyId);
+        },
+        [](const AtomString&) {
+            return true;
+        }
+    );
+}
+
+bool isAccelerated(const AnimatableCSSProperty& property, const Settings& settings)
+{
+    return WTF::switchOn(property,
+        [&](CSSPropertyID propertyId) {
+            return CSSProperty::animationIsAccelerated(propertyId, settings);
+        },
+        [](const AtomString&) {
+            return false;
+        }
+    );
+}
+
+bool canInterpolate(const AnimatableCSSProperty& property)
+{
+    return WTF::switchOn(property,
+        [](CSSPropertyID propertyId) {
+            return propertyId == CSSPropertyCustom || !!WrapperMap::singleton().wrapper(propertyId);
+        },
+        [](const AtomString&) {
+            // FIXME: this should only be true for property that are registered custom properties.
```

Comment:
```suggestion
            // FIXME: This should only be true for properties that are registered custom properties.
```
But also, I’m not sure I understand the meaning?

Reply:
Me either. (to be clear, its an existing comment)